### PR TITLE
Add field in the sidebar widget to filter shows by related post

### PR DESCRIPTION
**Pull request by borkweb**
This combines the work done by @thomasdebruin (with a few extra tweaks) to add the ability to only display shows that are related to a specific post.

🎥 https://d.pr/v/VFkDi4

### CODE REVIEW

File: output/gigpress_sidebar.php [hunk 1]
```diff
@@ -164,6 +166,25 @@ function form( $instance ) {
 			</select>
 		</p>
 
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'related' ) ); ?>">
+				<?php _e( 'Only display shows related to this post', 'gigpress' ); ?>
```
Comment (lelandf) [inline output/gigpress_sidebar.php:171]:
```suggestion
				<?php esc_html_e( 'Only display shows related to this post', 'gigpress' ); ?>
```
Should translation strings like this be escaped?

**Review by skyshab — APPROVED** [review]
you the man

File: output/gigpress_sidebar.php [hunk 2]
```diff
@@ -164,6 +166,25 @@ function form( $instance ) {
...
+			</label>
+			<select style="width:100%;" id="<?php echo esc_attr( $this->get_field_id( 'related' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'related' ) ); ?>">
+				<option value="">--</option>
+				<?php
+				$posts = fetch_gigpress_related_posts();
+				if ( $posts != false ) :
+					foreach ( $posts as $this_post ) :
+						?>
+						<option value="<?php echo absint( $this_post->ID ); ?>" <?php echo ( $related == $this_post->ID ) ? 'selected="selected"' : ''; ?>>
```
Comment (lelandf) [inline output/gigpress_sidebar.php:180]: `$related == $this_post->ID`

Can we "triple equals" this to be more strictly typed?

Might require some type coercion just to make sure they're both integers.

**Review by Camwyn — APPROVED** [review]
looks good!

File: output/gigpress_sidebar.php (hunk 2 above)
Comment (lelandf) [inline output/gigpress_sidebar.php:180]: `absint( $this_post->ID )`

I wonder if [esc_attr](https://developer.wordpress.org/reference/functions/esc_attr/) would be a more appropriate escaping function here?

Probably doesn't matter as absint seems perfectly effective to me in this case, just leaving a note.

Comment (borkweb) [reply output/gigpress_sidebar.php:180]: When dealing with integers, absint and esc_attr are pretty interchangeable, with absint being the more restrictive one.

**Pull request merged**